### PR TITLE
MGMT-1376 Add repository images list to the assisted-installer.yaml

### PR DESCRIPTION
**Pull request by eranco74**

### CODE REVIEW

File: release/main.py [hunk 1]
```diff
@@ -35,20 +39,69 @@ def tag_all(manifest, tag, delete_if_exists=False):
...
+    repos_with_tag = []
+    for repo in manifest_contnet.keys():
+        if gtools.tag_exists(repo, tag):
+            repos_with_tag.append(repo)
+    if repos_with_tag:
+        for repo in repos_with_tag:
+            gtools.delete_tag(repo, tag)
+
+
+def check_images_exists(manifest, tag):
```
Comment (ronniel1) [inline release/main.py:64]: Notice that if we used the '--force' option than this validation is more complicated.
Its not enough to check that the tag exists. Need to verify that the SHA returned for the tag matches the SHA returned for the images with the git-hash tag.

File: release/main.py [hunk 2]
```diff
@@ -35,20 +39,69 @@ def tag_all(manifest, tag, delete_if_exists=False):
...
+                missing_images.append(image)
+    if missing_images:
+        raise Exception("Missinge images %s" % missing_images)
+
+
+def image_exists(image_name, tag):
+    '''Return True if image with the given tag exists in quay.io under ocpmetal namespace'''
+    url = IMAGES_URL_TEMPLATE.format(image_name, tag)
+    response = requests.get(url=url)
+    return response.ok
```
Comment (ronniel1) [inline release/main.py:86]: Should return the SHA of the image (see comment above)

File: release/main.py [hunk 3]
```diff
@@ -35,20 +39,69 @@ def tag_all(manifest, tag, delete_if_exists=False):
...
+        manifest_contnet = yaml.safe_load(manifest_file)
+    logger.info("Deleting %s tag from all repos", tag)
+    gtools = gittools.GitApiUtils()
+    repos_with_tag = []
+    for repo in manifest_contnet.keys():
+        if gtools.tag_exists(repo, tag):
+            repos_with_tag.append(repo)
+    if repos_with_tag:
+        for repo in repos_with_tag:
+            gtools.delete_tag(repo, tag)
```
Comment (ronniel1) [inline release/main.py:61]: Maybe output a message for each tag we delete

File: release/main.py [hunk 4]
```diff
@@ -35,20 +39,69 @@ def tag_all(manifest, tag, delete_if_exists=False):
...
     parser.add_argument('-m', '--manifest', help='Path to manifest file', type=str, default="./assisted-installer.yaml")
     args = parser.parse_args()
-    tag_all(args.manifest, args.tag, args.force)
-
+    if args.delete:
+      untag_all(args.manifest, args.tag)
+    elif args.check:
+        check_images_exists(args.manifest, args.tag)
+    else:
+        tag_all(args.manifest, args.tag, args.force)
```
Comment (ronniel1) [inline release/main.py:103]: Maybe also tag the assisted-installer-deployment repo with the same tag?

File: release/main.py (hunk 1 above)
Comment (eranco74) [reply release/main.py:64]: Thx, didn't think of that.

File: release/main.py (hunk 4 above)
Comment (eranco74) [reply release/main.py:120]: Another PR

**Pull request closed without merging**